meeteval/io/stm.py

```diff
@@ -66,55 +72,31 @@ def serialize(self):
         return (f'{self.filename} {self.channel} {self.speaker_id} '
                 f'{self.begin_time} {self.end_time} {self.transcript}')
 
-    def replace(self, **kwargs):
-        """
-        Return a new instance of the named tuple replacing specified fields with new values.
-
-        >>> line = STMLine.parse('rec1 0 A 10 20 Hello World')
-        >>> line
-        STMLine(filename='rec1', channel=0, speaker_id='A', begin_time=10, end_time=20, transcript='Hello World')
-        >>> line.replace(speaker_id='B')
-        STMLine(filename='rec1', channel=0, speaker_id='B', begin_time=10, end_time=20, transcript='Hello World')
-        """
-        return self._replace(**kwargs)
-
 
 @dataclass(frozen=True)
-class STM:
+class STM(Base):
     lines: List[STMLine]
+    line_cls = STMLine
 
     @classmethod
-    def load(cls, stm_file: [Path, str, tuple, list]) -> 'STM':
-        if isinstance(stm_file, (tuple, list)):
-            return STM.merge(*[STM.load(f) for f in stm_file])
-
-        with open(stm_file, 'r') as fd:
-            return cls([
-                STMLine.parse(line)
-                for line in fd
-                if len(line.strip()) > 0 and not line.strip().startswith(';')
-            ])
-
-    def _repr_pretty_(self, p, cycle):
-        name = self.__class__.__name__
-        with p.group(len(name) + 1, name + '(', ')'):
-            if cycle:
-                p.text('...')
-            elif len(self.lines):
-                p.pretty(list(self.lines))
+    def _load(cls, file_descriptor) -> 'List[STMLine]':
+        return [
+            STMLine.parse(line)
+            for line in file_descriptor
+            if len(line.strip()) > 0 and not line.strip().startswith(';')
+        ]
 
     @classmethod
     def merge(cls, *stms) -> 'STM':
```

Comment:
Does it make sense to move `merge` into the base class?

Reply:
I just kept merge, because it was already implemented. Is it somewhere used?

Reply:
I think not anymore. It was used in one of the `load` functions